zio-cli/shared/src/test/scala/zio/cli/OptionsSpec.scala

```diff
@@ -0,0 +1,79 @@
+package zio.cli
+
+import zio.test.Assertion._
+import zio.test._
+import zio.cli.HelpDoc.dsl.{ p, error }
+
+object OptionsSpec extends DefaultRunnableSpec {
+
+  val f: Options[String]            = Options.text("firstname")
+  val l: Options[String]            = Options.text("lastname")
+  val a: Options[BigInt]            = Options.integer("age")
+  val aOpt: Options[Option[BigInt]] = Options.integer("age").optional
+
+  val options = f :: l :: a
+
+  def spec = suite("Options Suite")(
+    testM("validate text option") {
+      val r = f.validate(List("--firstname", "John"), ParserOptions.default)
+      assertM(r)(equalTo(List() -> "John"))
+    },
+    testM("validate integer option") {
+      val r = a.validate(List("--age", "100"), ParserOptions.default)
+      assertM(r)(equalTo(List() -> BigInt(100)))
+    },
+    testM("validate option and get remainder") {
+      val r = f.validate(List("--firstname", "John", "--lastname", "Doe"), ParserOptions.default)
+      assertM(r)(equalTo(List("--lastname", "Doe") -> "John"))
+    },
+    
+    testM("validate option and get remainder with different ordering") {
+      val r = f.validate(List("--bar", "baz", "--firstname", "John", "--lastname", "Doe"), ParserOptions.default)
+      assertM(r)(equalTo(List("--bar", "baz", "--lastname", "Doe") -> "John"))
+    },
+
+    testM("validate when no valid values are passed") {
+      val r = f.validate(List("--lastname", "Doe"), ParserOptions.default)
+      val expected =
+        List(p(error("No option found!. Was expecting option: --firstname.")))
```

Comment:
This test is a bit fragile. I think we should test only that it fails, not what the message is.